canvas/services/event.py

```diff
@@ -52,6 +44,29 @@ def get_question_stats(question):
                 answers[answer] = 0
             answers[answer] += 1
 
+    users = []
+    progress_submission_statuses = [0, 0, 0]
+    student_submissions = submissions.filter(uqj__user__role="Student")
+    correct_submissions = student_submissions.filter(is_correct=True).values("uqj__user_id")
+    for correct in correct_submissions:
+        if not users.__contains__(correct["uqj__user_id"]):
+            progress_submission_statuses[0] += 1
+            users.append(correct["uqj__user_id"])
+
+    partially_correct_submissions = student_submissions.filter(is_partially_correct=True).values("uqj__user_id")
+    for partial in partially_correct_submissions:
+        if not users.__contains__(partial["uqj__user_id"]):
+            progress_submission_statuses[1] += 1
+            users.append(partial["uqj__user_id"])
+
+    incorrect_submissions = student_submissions.filter(is_partially_correct=False, is_correct=False).values(
+        "uqj__user_id"
+    )
+    for incorrect in incorrect_submissions:
+        if not users.__contains__(incorrect["uqj__user_id"]):
```

Comment:
Google this a tad but I think the `in` operator is what `__contains__` means.
Like https://www.online-python.com/d8HlpBGn2K

The methods with `__something__` structure are called "dunders" and they typically correlate with a keyword/symbol (i.e. `__getattribute__` corresponds to the `.` operator)

tldr; I think `in` should be used, not `__contains__`